app/controllers/v0/sessions_controller.rb

```diff
@@ -33,6 +33,9 @@ def saml_callback
       if @saml_response.is_valid? && persist_session_and_user
         async_create_evss_account(@current_user)
         redirect_to SAML_CONFIG['relay'] + '?token=' + @session.token
+
+        obscure_token = Session.obscure_token(@session.token)
+        Rails.logger.info("Logged in user with id #{@session.uuid}, token #{obscure_token}")
```

Comment:
> the whole point of the thing is to be able to look up requests related to a user

I think I need more understanding of the requirement.

If it's, "_we need to know what user **abcd-1234** does when they log in_" then using a key of `sha26(@session.uuid)` will work just as good as `@session.uuid` with the added benefit of not leaking the real `uuid`. 

If it's, "_we need to know what user **William Peter Ryan** does when they log in_", then this isn't possible no matter what we do because we do not store such PII permanently.

@jcmeloni I don't see why we'd ever encrypt instead of just hash, then again I probably need more info on the requirement.